### PR TITLE
[FR] Add multipage Business card test and update documentation

### PR DESCRIPTION
**Pull request by maririos**
It also calculates the `Page` property for the field `ContactNames` as this is not returned by the service

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/README.md [hunk 1]
```diff
@@ -204,7 +207,7 @@ foreach (RecognizedForm form in forms)
 ```
 
 ### Recognize Receipts
-Recognize data from US sales receipts using a prebuilt model.
+Recognize data from US sales receipts using a prebuilt model. Receipt fields recognized by the service can be found [here][service_recognize_receipt_fields].
```
Comment (kristapratico) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:210]:
```suggestion
Recognize data from sales receipts using a prebuilt model. Receipt fields recognized by the service can be found [here][service_recognize_receipt_fields].
```

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs [hunk 2]
```diff
@@ -163,5 +162,29 @@ private static FormElement ResolveTextReference(IReadOnlyList<ReadResult> readRe
...
             throw new InvalidOperationException($"Failed to parse element reference: {reference}");
         }
+
+        /// <summary>
+        /// Business Cards pre-built model doesn't return a page number for the `ContactNames` field.
+        /// This function looks into the FieldValue_internal to see if it corresponds to
+        /// `ContactNames` and verifies that the page value before returning it.
+        /// </summary>
+        /// <returns>Page value if the field is `ContactNames` for Business cards. If not, defaults to 1.</returns>
+        private static int CalculatePage(FieldValue_internal field)
```
Comment (kristapratico) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs:172]: what if instead of running this check for every FormField, we pass down a flag from the business card method that indicates that this is a business card and we do need to calculate the page for ContactNames?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/README.md (hunk 1 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:210]: I made the modifications in https://github.com/Azure/azure-sdk-for-net/pull/16026
Thanks!

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs (hunk 2 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs:172]: Good idea. I will try that

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs:172]: Updated

**Review by kristapratico — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 3]
```diff
@@ -1121,6 +1124,68 @@ public void StartRecognizeBusinessCardsFromUriThrowsForNonExistingContent()
...
+                    operation = await client.StartRecognizeBusinessCardsAsync(stream, options);
+                }
+            }
+            else
+            {
+                var uri = FormRecognizerTestEnvironment.CreateUri(TestFile.BusinessMultipage);
+                operation = await client.StartRecognizeBusinessCardsFromUriAsync(uri, options);
+            }
+
+            RecognizedFormCollection recognizedForms = await operation.WaitForCompletionAsync(PollingInterval);
```
Comment (iscai-msft) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:1150]: out of curiosity, since I think this code works: where is the `PollingInterval` variable defined? Is this a .net thing I'm not familiar with, I'm not seeing it defined anywhere in the test file, but a bunch of tests are using it with no problem

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample12_RecognizeBusinessCardsFromFile.cs [hunk 4]
```diff
@@ -73,40 +73,6 @@ public async Task RecognizeBusinessCardsFromFile()
...
-                            foreach (FormField departmentField in departmentFields.Value.AsList())
-                            {
-                                if (departmentField.Value.ValueType == FieldValueType.String)
-                                {
-                                    string department = departmentField.Value.AsString();
-
-                                    Console.WriteLine($"  Department: '{department}', with confidence {departmentField.Confidence}");
-                                }
-                            }
-                        }
```
Comment (iscai-msft) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample12_RecognizeBusinessCardsFromFile.cs:107]: another question of curiosity: why did you remove some fields and how did you decide which fields to remove? I assume you removed them since the sample is so long, but am just curious

**Review by iscai-msft — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs (hunk 3 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:1150]: Yes. We have it in the `FormRecognizerLiveTestBase` which this class inherits from. 
The [PollingInterval](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Infrastructure/FormRecognizerLiveTestBase.cs#L14) is our way to tell the tests that if we are in `Playback` mode don't wait.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample12_RecognizeBusinessCardsFromFile.cs (hunk 4 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample12_RecognizeBusinessCardsFromFile.cs:107]: Yes. too long and the readme was getting sooooo long. I am following what Java did, where for the readme snippets show only a few but for the real BusinessCards I am leaving them all so the user can just copy paste

**Review by samvaity — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/README.md [hunk 5]
```diff
@@ -291,6 +294,78 @@ using (FileStream stream = new FileStream(receiptPath, FileMode.Open))
 }
 ```
 
+### Recognize Business Cards
```
[the remark this answers is not in the document]
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:297]: Hmm great question. Specially with invoices coming, the more prebuilt things the more this readme is going to grow.
I think for this preview I will leave it as is, but once we have more prebuilt models we should reconsider

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs (hunk 2 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs:172]: >this behavior only exists on "ContactNames" field?

Yes.

>Is this a service bug that they are planning to update later?

It is not a service bug. We are asking them to consider adding the `Page` field so we don't have to do all of this. But they haven't agreed. Conversation is here => https://teams.microsoft.com/l/message/19:88f2d9dea2344075b5d4bc34d82a3d1c@thread.skype/1602281344886?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1602281344886&teamName=Azure%20SDK&channelName=Service%20-%20Form%20Recognizer&createdTime=1602281344886

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs:172]: I updated the description of the PR to include this change too.
